rust/src/storage/items/kb.rs

```diff
@@ -0,0 +1,333 @@
+// SPDX-FileCopyrightText: 2025 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
+
+//! Defines an KB item with its corresponding key in storage.
+
+use std::{collections::HashMap, fmt::Display, hash::Hash, net::IpAddr};
+
+use crate::storage::{ScanID, Target};
+
+#[derive(Eq, Debug, Clone)]
+#[cfg_attr(
+    feature = "serde_support",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "snake_case")
+)]
+/// List of keys for the KnowledgeBase
+pub enum KbKey {
+    // SSL/TLS
+    SslCert,
```

Comment:
Wouldn't it make more sense to create subkeys? Like
```
pub enum KbSSLKey  {
    Cert,
    Key,
    Password,
    CA,
}

pub enum KbKey {
   SSL(KbSSLKey),
   ...
}
```

instead of comments and naming? Especially when using a LSP lookup in a smaller window it can get frustrating when there are too many SSL***, Port*** entries which you have to skip through although you know the context.